src/test/java/com/commercetools/util/ArgumentsUtilsTest.java

```diff
@@ -0,0 +1,107 @@
+                .withMessage("String argument is expected to be non-blank");
+
+        assertThatIllegalArgumentException().isThrownBy(() -> requireNonBlank("\t\n\r"))
+                .withMessage("String argument is expected to be non-blank");
+    }
+
+    @Test
+    public void requireNonBlank_returnsValidArgument() throws Exception {
+        assertThat(requireNonBlank("hello")).isEqualTo("hello");
+        assertThat(requireNonBlank("   world   ")).isEqualTo("   world   ");
+        assertThat(requireNonBlank("\t\r!\n\f")).isEqualTo("\t\r!\n\f");
+
+        assertThat(requireNonBlank("hello", "message is ignored")).isEqualTo("hello");
+        assertThat(requireNonBlank("   world   ", "message is ignored")).isEqualTo("   world   ");
+        assertThat(requireNonBlank("\t\r!\n\f", "message is ignored")).isEqualTo("\t\r!\n\f");
+    }
+
+    @Test
+    public void requireNonEmpty_throwsException_withCustomMessage() throws Exception {
+        assertThatIllegalArgumentException().isThrownBy(() -> requireNonEmpty(null, "must be not null"))
+                .withMessage("must be not null");
+
+        assertThatIllegalArgumentException().isThrownBy(() -> requireNonEmpty(new ArrayList<>(), "must be not empty"))
+                .withMessage("must be not empty");
+
+        assertThatIllegalArgumentException().isThrownBy(() -> requireNonEmpty(new HashSet<>(), "must be not empty"))
+                .withMessage("must be not empty");
+
+        assertThatIllegalArgumentException().isThrownBy(() -> requireNonEmpty(emptyList(), "must be not empty"))
+                .withMessage("must be not empty");
+
+        assertThatIllegalArgumentException().isThrownBy(() -> requireNonEmpty(emptySet(), "must be not empty"))
+                .withMessage("must be not empty");
+    }
+
+    @Test
+    public void requireNonEmpty_throwsException_withDefaultMessage() throws Exception {
+        assertThatIllegalArgumentException().isThrownBy(() -> requireNonEmpty(null))
+                .withMessage("Collection argument is expected to be non-empty");
+
+        assertThatIllegalArgumentException().isThrownBy(() -> requireNonEmpty(new ArrayList<>()))
+                .withMessage("Collection argument is expected to be non-empty");
+
+        assertThatIllegalArgumentException().isThrownBy(() -> requireNonEmpty(new HashSet<>()))
+                .withMessage("Collection argument is expected to be non-empty");
+
+        assertThatIllegalArgumentException().isThrownBy(() -> requireNonEmpty(emptyList()))
+                .withMessage("Collection argument is expected to be non-empty");
+
+        assertThatIllegalArgumentException().isThrownBy(() -> requireNonEmpty(emptySet()))
+                .withMessage("Collection argument is expected to be non-empty");
+    }
+
+    @Test
+    public void requireNonEmpty_returnsTheCollection() throws Exception {
+        HashSet<String> testSet = new HashSet<>(asList("x", "y", "z"));
```

Comment:
Also add Collection with one null value?

Reply:
collection with one _null_ value is not an empty collection - it contains one reference. It is not a case for this utility method (the method does not validate content)